builder/virtualbox/iso/step_create_disk.go

```diff
@@ -116,45 +76,88 @@ func (s *stepCreateDisk) Run(ctx context.Context, state multistep.StateBag) mult
 		}
 	}
 
-	// Attach the disk to the controller
-	controllerName := "IDE Controller"
-	if config.HardDriveInterface == "sata" {
-		controllerName = "SATA Controller"
-	} else if config.HardDriveInterface == "scsi" {
-		controllerName = "SCSI Controller"
-	} else if config.HardDriveInterface == "virtio" {
-		controllerName = "VirtIO Controller"
-	} else if config.HardDriveInterface == "pcie" {
-		controllerName = "NVMe Controller"
-	}
+	format := "VDI"
 
-	nonrotational := "off"
-	if config.HardDriveNonrotational {
-		nonrotational = "on"
-	}
+	// The main disk and additional disks
+	diskFullPaths := []string{}
+	diskSizes := []uint{config.DiskSize}
+	if len(config.AdditionalDiskSize) == 0 {
+		// If there are no additional disks, use disk naming as before
+		diskFullPaths = append(diskFullPaths, filepath.Join(config.OutputDir, fmt.Sprintf("%s.%s", config.VMName, strings.ToLower(format))))
+	} else {
+		// If there are additional disks, use consistent naming with numbers
+		diskFullPaths = append(diskFullPaths, filepath.Join(config.OutputDir, fmt.Sprintf("%s-0.%s", config.VMName, strings.ToLower(format))))
 
-	discard := "off"
-	if config.HardDriveDiscard {
-		discard = "on"
+		for i, diskSize := range config.AdditionalDiskSize {
+			diskPath := filepath.Join(config.OutputDir, fmt.Sprintf("%s-%d.%s", config.VMName, i+1, strings.ToLower(format)))
+			diskFullPaths = append(diskFullPaths, diskPath)
+			diskSizes = append(diskSizes, diskSize)
+		}
 	}
 
-	for i := range diskFullPaths {
-		command := []string{
-			"storageattach", vmName,
-			"--storagectl", controllerName,
-			"--port", strconv.FormatUint(uint64(i), 10),
-			"--device", "0",
-			"--type", "hdd",
-			"--medium", diskFullPaths[i],
-			"--nonrotational", nonrotational,
-			"--discard", discard,
+	if path.Ext(state.Get("iso_path").(string)) != ".vhd" {
```

Comment:
Hello @lbajolet-hashicorp ,
Per [documentation](https://developer.hashicorp.com/packer/integrations/hashicorp/virtualbox/latest/components/builder/iso#required:) and per [code](https://github.com/hashicorp/packer-plugin-virtualbox/blob/dde0385c69e984fb267062999c7b7a21ec398e6e/builder/virtualbox/iso/builder.hcl2spec.go#L28) we are sure that the `iso_url` is always set (in case of iso builder).